### PR TITLE
Enabled subscriptions over dlq topic

### PR DESCRIPTION
**Pull request by rutul25**

### CODE REVIEW

File: internal/topic/helper.go [hunk 1]
```diff
@@ -13,3 +14,8 @@ const (
 func GetTopicName(projectID string, name string) string {
 	return fmt.Sprintf(TopicNameFormat, projectID, name)
 }
+
+// IsTopicDLQ helper checks if the topic is dlq topic
+func IsTopicDLQ(topicName string) bool {
```
Comment (vnktram) [inline internal/topic/helper.go:19]: nit: IsTopicDLQ -> IsDLQTopic

File: internal/subscription/core.go [hunk 2]
```diff
@@ -81,32 +81,39 @@ func (c *Core) CreateSubscription(ctx context.Context, m *Model) error {
...
-		}
+	// this creates the needed delay topics in the broker
+	err = createDelayTopics(ctx, m, c.topicCore)
+	if err != nil {
+		logger.Ctx(ctx).Errorw("failed to create delay topics", "error", err.Error())
+		return err
+	}
+
+	// for subscription over deadletter topics, skip the deadletter topic creation
+	if !topicModel.IsDeadLetterTopic() {
```
Comment (vnktram) [inline internal/subscription/core.go:116]: Add a label to the subscription denoting that this is a DLQ subscription.

**Review by vnktram — CHANGES_REQUESTED** [review]
LGTM. Minor changes requested.

**Review by vnktram — APPROVED** [review]
LGTM

**Review by vnktram — APPROVED** [review]
LGTM. Please add unit tests to ensure that no DLQ is created for the subscription.

File: internal/subscription/core.go [hunk 3]
```diff
@@ -112,6 +113,8 @@ func (c *Core) CreateSubscription(ctx context.Context, m *Model) error {
 		return err
 	}
 
+	fmt.Println(topicModel)
```
Comment (vnktram) [inline internal/subscription/core.go:116]: Why are we using fmt.Println() and not logger.Ctx(ctx).Infow()

File: internal/subscription/core_test.go [hunk 4]
```diff
@@ -9,10 +9,119 @@ import (
...
+
+func TestSubscriptionCore_CreateSubscription_DLQTopic(t *testing.T) {
+	ctx := context.Background()
+	ctrl := gomock.NewController(t)
+	mockProjectCore := pCore.NewMockICore(ctrl)
+	mockTopicCore := tCore.NewMockICore(ctrl)
+	mockRepo := repo.NewMockIRepo(ctrl)
+
+	core := NewCore(mockRepo, mockProjectCore, mockTopicCore)
+	sub := Model{
```
Comment (vnktram) [inline internal/subscription/core_test.go:81]: Use a getModel helper function instead.

**Review by vnktram — CHANGES_REQUESTED** [review]
Please address comments

File: internal/subscription/core.go (hunk 3 above)
Comment (rutul25) [reply internal/subscription/core.go:116]: Removing this. Added for local testing.

**Review by vnktram — APPROVED** [review]
LGTM

**Pull request merged**